src/modules/openstack.rs

```diff
@@ -0,0 +1,133 @@
+use yaml_rust::YamlLoader;
+
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::openstack::OspConfig;
+use crate::formatter::StringFormatter;
+use crate::utils;
+
+type Cloud = String;
+type Project = String;
+
+fn get_osp_project_from_config(context: &Context, osp_cloud: Option<&str>) -> Option<Project> {
+    let config_local = context.get_env("PWD").unwrap() + "/clouds.yaml";
+    let config_user = dirs_next::home_dir()?.join(".config/openstack/clouds.yaml");
+    let config_global = "/etc/openstack/clouds.yaml";
+    // Attempt to follow OpenStack standards for clouds.yaml location:
+    // 1st = $PWD/clouds.yaml, 2nd = $HOME/.config/openstack/clouds.yaml, 3rd = /etc/openstack/clouds.yaml
+    // TODO: I'm sure there is a better way to do this.
```

Comment:
You could try using an iterator and a [find_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find_map) here? You could build an array of possible paths, then iterate through them using the find_map to grab the first one with content.

Reply:
Fixed. Please review.